compiler/rustc_errors/src/lib.rs

```diff
@@ -892,6 +910,16 @@ impl HandlerInner {
         self.delay_as_bug(diagnostic)
     }
 
+    #[track_caller]
+    fn delay_good_path_bug(&mut self, msg: &str) {
+        let mut diagnostic = Diagnostic::new(Level::Bug, msg);
+        if self.flags.report_delayed_bugs {
+            self.emit_diagnostic(&diagnostic);
+        }
+        diagnostic.note(&format!("delayed at {}", std::backtrace::Backtrace::force_capture()));
```

Comment:
This might need to account for formatting, especially around newlines. But also, is this necessary? If you can force the above `emit_diagnostic` to run, wouldn't that give you a backtrace (and a query stack, etc.)? Or is this for diagnosing issues on CI?

Reply:
The purpose of the backtrace here is to capture the full call site, but to not print anything, because it is not yet an error, so forcing the call to `emit_diagnostic` would not serve this purpose. It popped in CI when I was working on the PR and the formatting worked out alright.